### PR TITLE
[inductor] Refactor fallback kernel handling

### PR DESCRIPTION
**Pull request by jansel**

### CODE REVIEW

File: tests/test_torchinductor.py [hunk 1]
```diff
@@ -60,6 +61,21 @@
 torchinductor.config.triton.autotune = False  # too slow
 
 
+def requires_decomp(fn):
+    """Decorator to disable test of a decomp is missing"""
```
Comment (voznesenskym) [inline tests/test_torchinductor.py:65]: nit: if a decomp is missing?

File: torchinductor/decomposition.py [hunk 2]
```diff
@@ -250,3 +252,13 @@ def masked_fill_(x, mask, value):
 @register_decomposition([aten.log1p])
 def log1p(x):
     return torch.log(x + 1)
+
+
+@register_decomposition([aten.baddbmm])
+def baddbmm(self, batch1, batch2, beta=1, alpha=1):
+    result = torch.bmm(batch1, batch2)
+    if not isinstance(alpha, (int, float)) or alpha != 1:
```
Comment (voznesenskym) [inline torchinductor/decomposition.py:260]: nit: I think this can be `isinstance(alpha, numbers.Number)`

same below.

File: torchinductor/ir.py [hunk 3]
```diff
@@ -1577,6 +1577,12 @@ def copy_input(x):
 
     @classmethod
     def realize_input(cls, x):
+        if x is None:
+            return V.graph.add_tensor_constant(torch.tensor(()))
```
Comment (voznesenskym) [inline torchinductor/ir.py:1581]: Am pretty new to understanding the IR, so apologies if this is a dumb question - why does None make a default tensor constant?

Comment (jansel) [reply torchinductor/ir.py:1581]: This is a workaround in that some other code expects the graph to return only tensors.  The None values are coming from `aten.convolution_backwards` and are unused, so it doesn't matter.

Comment (voznesenskym) [reply torchinductor/ir.py:1581]: Understood! Would we eventually ever have a specialized "NoneTensor" or something like that, to signal that this came from a None?

File: torchinductor/lowering.py [hunk 4]
```diff
@@ -47,6 +47,15 @@
 }
 
 
+def has_torchvision_roi_align():
+    try:
+        from torchvision.ops import roi_align  # noqa
+
+        return roi_align is not None
+    except (ImportError, ModuleNotFoundError):
+        return False
```
Comment (voznesenskym) [inline torchinductor/lowering.py:56]: I wonder if importlib can help avoid exceptions for control flow here, no strong opinions though.

**Review by voznesenskym — APPROVED** [review]

File: torchinductor/ir.py (hunk 3 above)
Comment (jansel) [reply torchinductor/ir.py:1581]: I'm not convinced None carries any useful signal we want to preserve in the backend.  In Dynamo, None is useful, but in the backend you can't do anything with None other than ignore it.

**Pull request merged**